frappe/model/document.py

```diff
@@ -1506,16 +1506,18 @@ def unlock(self):
 		if self in frappe.local.locked_documents:
 			frappe.local.locked_documents.remove(self)
 
-	# validation helpers
-	def validate_from_to_dates(self, from_date_field, to_date_field):
-		"""
-		Generic validation to verify date sequence
-		"""
-		if date_diff(self.get(to_date_field), self.get(from_date_field)) < 0:
+	def validate_from_to_dates(self, from_date_field: str, to_date_field: str) -> None:
+		"""Validate that the value of `from_date_field` is not later than the value of `to_date_field`."""
+		from_date = self.get(from_date_field)
+		to_date = self.get(to_date_field)
+		if not (from_date and to_date):
+			return
```

Comment:
this looks breaking in behaviour? :eyes:  - if only end date was present, then this api would've tested against current date vs the given end date in order to ensure the end date was not of past - with this changed api this wont happen and would just skip the validation 

imo we should only validate if end date or both start and end date are given otherwise we should just skip the validation?

1. start and end are there - validate
2. only start - doesn't make sense for validation
3. only end - validate (for the above mentioned reason)

Reply:
Yes, it is definitely breaking, but I don't think the old behavior was intended/know.

Why do you think it would make sense to implicitly validate that `to_date` is in the future? (It also used to validate that `start_date` is in the past.)

Reply:
There is no point in putting a end date if there is no start date present (?) 
So if someone puts (only) past enddate it should raise an error is what I'm thinking 🤔

Start date can be in future or past so it doesn't really matter if we validate it or not

Reply:
This method compares the values of two fields and should ensure that from_date is not earlier than to_date. This validation doesn't make sense if one date is empty.

If I want from_date to default to today, I can achieve that in other ways.

To me it feels like unexpected, unintended behaviour that we implicitly assume an empty date field to hold the current date.